bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -29,6 +33,9 @@ impl Lambda {
     }
 
     pub fn set_init_duration_metric(&self, init_duration_ms: f64) -> Result<(), errors::Insert> {
+        if !self.config.enhanced_metrics {
```

Comment:
Should we use Option<Error> instead of result with unit?

Reply:
not sure what you mean by result with unit?

Reply:
Unit type is (). Since it doesn't hold anything, in this case Some(Err)/None can be used